CHANGELOG.md

```diff
@@ -6,6 +6,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [9.0.0]
+### Changed
+- Bump devDependency `typescript` from `~4.8.4` to `~5.0.4` ([#185](https://github.com/MetaMask/utils/pull/185)).
+- Bump dependency `semver` from `^5.7.1` to `^7.6.0` ([#181](https://github.com/MetaMask/utils/pull/181)).
+
+### Fixed
+- **BREAKING:** Use `ts-bridge` for building, which supports entry points `@metamask/utils` and `@metamask/utils/node` for both CJS and ESM, and `@metamask/utils/package.json` for JSON. ([#182](https://github.com/MetaMask/utils/pull/182)).
+  - **BREAKING:** Exposes separate build entry points and type declarations for CommonJS and ESM via package manifest `exports`.
+  - **BREAKING:** It's no longer possible to import files from the dist folder directly.
```

Comment:
Am I understanding these changes correctly? Suggestions welcome on better wording etc.

Reply:
@Mrtenz @mcmire

Reply:
This was already not possible in `8.4.0`. There's no breaking change here.

Reply:
Ah, in that case would this work? Or are you suggesting moving the top-level bullet point?

```suggestion
- **BREAKING:** Use `ts-bridge` for building, which supports entry points `@metamask/utils` and `@metamask/utils/node` for both CommonJS and ESM, and `@metamask/utils/package.json` for JSON. ([#182](https://github.com/MetaMask/utils/pull/182)).
  - **BREAKING:** Exposes separate build entry points and type declarations for CommonJS and ESM via package manifest `exports`.
```

Reply:
I don't think it needs to be mentioned in the changelog at all? There was an ESM build already, and the exports haven't changed.

Reply:
I think something is worth mentioning, but perhaps the reason behind the switch to `ts-bridge` is unclear.

Previously, the build process used a combination of `tsup` and `tsc` to produce CommonJS- and ESM-compatible JavaScript and type declaration files. There were two problems with this, however:

- Viewing the `dist` folder, and thus the published version of this package, revealed a set of "chunk" files. These were created by `tsup` from common pieces of code referenced across multiple files in order to make the package more tree-shakeable. However, because they were randomly named and because it was difficult to tell their contents at a glance, these chunk files made debugging our packages difficult for the extension team.
- Although the build process produced both CommonJS- and ESM-compatible variants of JavaScript files, it only produced a CommonJS-compatible variant of the type declaration files. Furthermore, this CJS variant is what the `exports` field in `package.json` referred to. This means that an ESM library written in TypeScript would likely not be able to use `@metamask/utils`, because TypeScript would try to import type declarations and realize that they were not ESM-compatible.

Switching to `ts-bridge` solves both of these problems. It removes the chunk files, it creates two variants of type declaration files in addition to JavaScript file, and it points to both variants of both types of files in `package.json`. In fact, this package should now pass all tests run by the ["Are the Types Wrong?"](https://arethetypeswrong.github.io/) tool.

Reply:
So, I agree that we don't need to provide a changelog entry for splitting type declarations into CJS and ESM variants, but perhaps we want one to say that we removed the chunk files?

Minimally we could have:



or, if we do want to include the bit about type declarations, we could have:

``` suggestion
- Remove chunk files ([#182](https://github.com/MetaMask/utils/pull/182))
  - 
- Produce and export ESM-compatible TypeScript type declaration files in addition to CommonJS-compatible declaration files ([#182](https://github.com/MetaMask/utils/pull/182))
  - Previously, this package shipped with only one variant of type declaration files, and these files were only CommonJS-compatible, and the `exports` field in `package.json` linked to these files. This is an anti-pattern and was rightfully flagged by the ["Are the Types Wrong?"](https://arethetypeswrong.github.io/) tool as ["masquerading as CJS"](https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/FalseCJS.md). This check now passes.